src/application-delegate.coffee

```diff
@@ -175,8 +174,38 @@ class ApplicationDelegate
 
   onUpdateAvailable: (callback) ->
     outerCallback = (event, message, detail) ->
-      if message is 'update-available'
-        callback(detail)
+      # TODO: Yes, this is strange that `onUpdateAvailable` is listening for
+      # `did-begin-downloading-update`. We currently have no mechanism to know
+      # if there is an update, so begin of downloading is a good proxy.
+      callback(detail) if message is 'did-begin-downloading-update'
```

Comment:
Note that this is changing the semantics of `onUpdateAvailable()`. It used to indicate that the update is downloaded and ready to be installed. Now, it will indicate that there is an update available, but before download and all that.